I'm curious to look into medal data from the 2024 Summer Olympics. First, I'll load the data and take a look at the first 10 entries.

In [1]:
import pandas as pd
paris2024 = pd.read_csv('./paris2024.csv')
paris2024.head(10)

NOC  Gold  Silver  Bronze  Total
0  USA    40      44      42    126
1  CHN    40      27      24     91
2  JPN    20      12      13     45
3  AUS    18      19      16     53
4  FRA    16      26      22     64
5  NED    15       7      12     34
6  GBR    14      22      29     65
7  KOR    13       9      10     32
8  ITA    12      13      15     40
9  GER    12      13       8     33

The data seems to be sorted by rank, which is based on the number of gold medals (then the number of silver medals, then the number of bronze medals). I'm curious to see what the list looks like when you sort by number of total medals instead.

In [2]:
paris_sorted_by_total = paris2024.sort_values('Total', ascending=False)
paris_sorted_by_total.head(10)

NOC  Gold  Silver  Bronze  Total
0  USA    40      44      42    126
1  CHN    40      27      24     91
6  GBR    14      22      29     65
4  FRA    16      26      22     64
3  AUS    18      19      16     53
2  JPN    20      12      13     45
8  ITA    12      13      15     40
5  NED    15       7      12     34
9  GER    12      13       8     33
7  KOR    13       9      10     32

The list has changed slightly, although the US and China are still at the top of the podium. I want to find out how many total medals were awarded so I can see what percentage of medals were won by each country.

In [3]:
total_number_medals = paris2024['Total'].sum()
total_number_medals

1041

So 1041 medals were given out in total. I'll use this to add a new column that lists the percentage of medals won by each country.

In [4]:
paris2024['Percent of Total Medals'] = paris2024['Total']/total_number_medals
paris2024

NOC  Gold  Silver  Bronze  Total  Percent of Total Medals
0   USA    40      44      42    126                 0.121037
1   CHN    40      27      24     91                 0.087416
2   JPN    20      12      13     45                 0.043228
3   AUS    18      19      16     53                 0.050913
4   FRA    16      26      22     64                 0.061479
..  ...   ...     ...     ...    ...                      ...
86  SIN     0       0       1      1                 0.000961
87  CIV     0       0       1      1                 0.000961
88  ROT     0       0       1      1                 0.000961
89  PER     0       0       1      1                 0.000961
90  SVK     0       0       1      1                 0.000961

[91 rows x 6 columns]

The US seems to have won over 12% of all medals! Let's take a look at the info about each column and see if there's any data missing.

In [5]:
paris2024.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 91 entries, 0 to 90
Data columns (total 6 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   NOC                      91 non-null     object 
 1   Gold                     91 non-null     int64  
 2   Silver                   91 non-null     int64  
 3   Bronze                   91 non-null     int64  
 4   Total                    91 non-null     int64  
 5   Percent of Total Medals  91 non-null     float64
dtypes: float64(1), int64(4), object(1)
memory usage: 4.4+ KB


There are 91 countries that won medals. No entries are missing. All the data types look consistent with the column. Now, I'm curious if there's an equal number of gold, silver, and bronze medals.

In [6]:
medals = ['Gold', 'Silver', 'Bronze']
for medal in medals:
    print(f'{medal}: ' + str(paris2024[medal].sum()))

Gold: 328
Silver: 330
Bronze: 383


Interestingly, there were many more bronze medals given out than gold and silver! I wonder if that means there were some ties. Now, let's look at data from the Tokyo 2020 Olympics.

In [7]:
tokyo2020 = pd.read_csv('./Tokyo2020.csv')
tokyo2020.head(10)

NOC  Gold  Silver  Bronze  Total
0  USA    39      41      33    113
1  CHN    38      32      19     89
2  JPN    27      14      17     58
3  GBR    22      20      22     64
4  ROC    20      28      23     71
5  AUS    17       7      22     46
6  NED    10      12      14     36
7  FRA    10      12      11     33
8  GER    10      11      16     37
9  ITA    10      10      20     40

I want to merge these two data sets so I can compare the two Olympics, but I need to rename the columns first so there aren't duplicate column names.

In [8]:
paris2024 = paris2024.rename({
    'Gold':'Gold 2024',
    'Silver':'Silver 2024',
    'Bronze':'Bronze 2024',
    'Total':'Total 2024',
    'Percent of Total Medals':'Percent of Total 2024'}, axis=1)
paris2024

NOC  Gold 2024  Silver 2024  Bronze 2024  Total 2024  \
0   USA         40           44           42         126   
1   CHN         40           27           24          91   
2   JPN         20           12           13          45   
3   AUS         18           19           16          53   
4   FRA         16           26           22          64   
..  ...        ...          ...          ...         ...   
86  SIN          0            0            1           1   
87  CIV          0            0            1           1   
88  ROT          0            0            1           1   
89  PER          0            0            1           1   
90  SVK          0            0            1           1   

    Percent of Total 2024  
0                0.121037  
1                0.087416  
2                0.043228  
3                0.050913  
4                0.061479  
..                    ...  
86               0.000961  
87               0.000961  
88               0.000961  
89               0.000961  
90               0.000961  

[91 rows x 6 columns]

I'm going to do the same thing to the Tokyo columns and add the percent of total column.

In [9]:
tokyo2020['Percent of Total 2020'] = tokyo2020['Total']/tokyo2020['Total'].sum()
tokyo2020 = tokyo2020.rename({
    'Gold':'Gold 2020',
    'Silver':'Silver 2020',
    'Bronze':'Bronze 2020',
    'Total':'Total 2020'}, axis=1)
tokyo2020

NOC  Gold 2020  Silver 2020  Bronze 2020  Total 2020  \
0   USA         39           41           33         113   
1   CHN         38           32           19          89   
2   JPN         27           14           17          58   
3   GBR         22           20           22          64   
4   ROC         20           28           23          71   
..  ...        ...          ...          ...         ...   
87  MDA          0            0            1           1   
88  CIV          0            0            1           1   
89  BOT          0            0            1           1   
90  GHA          0            0            1           1   
91  KUW          0            0            1           1   

    Percent of Total 2020  
0                0.105019  
1                0.082714  
2                0.053903  
3                0.059480  
4                0.065985  
..                    ...  
87               0.000929  
88               0.000929  
89               0.000929  
90               0.000929  
91               0.000929  

[92 rows x 6 columns]

Now, I'll merge them, putting Tokyo on the left and Paris on the right. I'm going to do an inner merge so I'm only comparing the countries that earned a medal in each Olympics.

In [10]:
olympics = pd.merge(
            left=tokyo2020,
            right=paris2024,
            left_on='NOC',
            right_on='NOC')
olympics

NOC  Gold 2020  Silver 2020  Bronze 2020  Total 2020  \
0   USA         39           41           33         113   
1   CHN         38           32           19          89   
2   JPN         27           14           17          58   
3   GBR         22           20           22          64   
4   AUS         17            7           22          46   
..  ...        ...          ...          ...         ...   
69  MEX          0            0            4           4   
70  GRN          0            0            1           1   
71  MDA          0            0            1           1   
72  CIV          0            0            1           1   
73  BOT          0            0            1           1   

    Percent of Total 2020  Gold 2024  Silver 2024  Bronze 2024  Total 2024  \
0                0.105019         40           44           42         126   
1                0.082714         40           27           24          91   
2                0.053903         20           12           13          45   
3                0.059480         14           22           29          65   
4                0.042751         18           19           16          53   
..                    ...        ...          ...          ...         ...   
69               0.003717          0            3            2           5   
70               0.000929          0            0            2           2   
71               0.000929          0            1            3           4   
72               0.000929          0            0            1           1   
73               0.000929          1            1            0           2   

    Percent of Total 2024  
0                0.121037  
1                0.087416  
2                0.043228  
3                0.062440  
4                0.050913  
..                    ...  
69               0.004803  
70               0.001921  
71               0.003842  
72               0.000961  
73               0.001921  

[74 rows x 11 columns]

I'm curious to see which countries got more medals in 2024 than in 2020. I'll make a column of Booleans for this.

In [11]:
olympics['More Medals in 2024'] = olympics['Total 2024'] > olympics['Total 2020']
olympics

NOC  Gold 2020  Silver 2020  Bronze 2020  Total 2020  \
0   USA         39           41           33         113   
1   CHN         38           32           19          89   
2   JPN         27           14           17          58   
3   GBR         22           20           22          64   
4   AUS         17            7           22          46   
..  ...        ...          ...          ...         ...   
69  MEX          0            0            4           4   
70  GRN          0            0            1           1   
71  MDA          0            0            1           1   
72  CIV          0            0            1           1   
73  BOT          0            0            1           1   

    Percent of Total 2020  Gold 2024  Silver 2024  Bronze 2024  Total 2024  \
0                0.105019         40           44           42         126   
1                0.082714         40           27           24          91   
2                0.053903         20           12           13          45   
3                0.059480         14           22           29          65   
4                0.042751         18           19           16          53   
..                    ...        ...          ...          ...         ...   
69               0.003717          0            3            2           5   
70               0.000929          0            0            2           2   
71               0.000929          0            1            3           4   
72               0.000929          0            0            1           1   
73               0.000929          1            1            0           2   

    Percent of Total 2024  More Medals in 2024  
0                0.121037                 True  
1                0.087416                 True  
2                0.043228                False  
3                0.062440                 True  
4                0.050913                 True  
..                    ...                  ...  
69               0.004803                 True  
70               0.001921                 True  
71               0.003842                 True  
72               0.000961                False  
73               0.001921                 True  

[74 rows x 12 columns]

So most of the top 5 got more medals in 2024 than in 2020. I'll now calculate what percentage of countries improved.

In [12]:
olympics['More Medals in 2024'].mean()

0.40540540540540543

That tells me that only around 40.54% of the countries in this dataset improved their total medal count. I'm now curious to find out medals per capita. To do this, I'll have to use a population dataset with country codes.

In [13]:
populations = pd.read_csv('./Population.csv')
populations.head()

Rank CCA3 Country/Territory           Capital Continent  2022 Population  \
0    36  AFG       Afghanistan             Kabul      Asia         41128771   
1   138  ALB           Albania            Tirana    Europe          2842321   
2    34  DZA           Algeria           Algiers    Africa         44903225   
3   213  ASM    American Samoa         Pago Pago   Oceania            44273   
4   203  AND           Andorra  Andorra la Vella    Europe            79824   

   2020 Population  2015 Population  2010 Population  2000 Population  \
0         38972230         33753499         28189672         19542982   
1          2866849          2882481          2913399          3182021   
2         43451666         39543154         35856344         30774621   
3            46189            51368            54849            58230   
4            77700            71746            71519            66097   

   1990 Population  1980 Population  1970 Population  Area (km²)  \
0         10694796         12486631         10752971      652230   
1          3295066          2941651          2324731       28748   
2         25518074         18739378         13795915     2381741   
3            47818            32886            27075         199   
4            53569            35611            19860         468   

   Density (per km²)  Growth Rate  World Population Percentage  
0            63.0587       1.0257                         0.52  
1            98.8702       0.9957                         0.04  
2            18.8531       1.0164                         0.56  
3           222.4774       0.9831                         0.00  
4           170.5641       1.0100                         0.00

I'm going to rename the 'CCA3' Column so it matches the column name of the olympics dataset.

In [14]:
populations = populations.rename({'CCA3':'NOC'}, axis=1)
populations.head()

Rank  NOC Country/Territory           Capital Continent  2022 Population  \
0    36  AFG       Afghanistan             Kabul      Asia         41128771   
1   138  ALB           Albania            Tirana    Europe          2842321   
2    34  DZA           Algeria           Algiers    Africa         44903225   
3   213  ASM    American Samoa         Pago Pago   Oceania            44273   
4   203  AND           Andorra  Andorra la Vella    Europe            79824   

   2020 Population  2015 Population  2010 Population  2000 Population  \
0         38972230         33753499         28189672         19542982   
1          2866849          2882481          2913399          3182021   
2         43451666         39543154         35856344         30774621   
3            46189            51368            54849            58230   
4            77700            71746            71519            66097   

   1990 Population  1980 Population  1970 Population  Area (km²)  \
0         10694796         12486631         10752971      652230   
1          3295066          2941651          2324731       28748   
2         25518074         18739378         13795915     2381741   
3            47818            32886            27075         199   
4            53569            35611            19860         468   

   Density (per km²)  Growth Rate  World Population Percentage  
0            63.0587       1.0257                         0.52  
1            98.8702       0.9957                         0.04  
2            18.8531       1.0164                         0.56  
3           222.4774       0.9831                         0.00  
4           170.5641       1.0100                         0.00

I'm going to get rid of some columns here that I won't use. This dataset only has information about 2020 and not 2024, so I'll only perform analysis with the 2020 data.

In [15]:
populations = populations[['NOC','Country/Territory','Continent','2020 Population']]
populations.head()

NOC Country/Territory Continent  2020 Population
0  AFG       Afghanistan      Asia         38972230
1  ALB           Albania    Europe          2866849
2  DZA           Algeria    Africa         43451666
3  ASM    American Samoa   Oceania            46189
4  AND           Andorra    Europe            77700

In [16]:
data2020 = pd.merge(left=tokyo2020, right=populations, left_on='NOC', right_on='NOC')
data2020.head()

NOC  Gold 2020  Silver 2020  Bronze 2020  Total 2020  \
0  USA         39           41           33         113   
1  CHN         38           32           19          89   
2  JPN         27           14           17          58   
3  GBR         22           20           22          64   
4  AUS         17            7           22          46   

   Percent of Total 2020 Country/Territory      Continent  2020 Population  
0               0.105019     United States  North America        335942003  
1               0.082714             China           Asia       1424929781  
2               0.053903             Japan           Asia        125244761  
3               0.059480    United Kingdom         Europe         67059474  
4               0.042751         Australia        Oceania         25670051

Now I've merged the Tokyo 2020 data with the population data, so I can see each country's medals per capita! Although, given how big the populations are, I'll do medals per 10,000,000 people.

In [17]:
data2020['Medals Per 10 Mil People'] = (data2020['Total 2020']/data2020['2020 Population'])*10000000
data2020.head()

NOC  Gold 2020  Silver 2020  Bronze 2020  Total 2020  \
0  USA         39           41           33         113   
1  CHN         38           32           19          89   
2  JPN         27           14           17          58   
3  GBR         22           20           22          64   
4  AUS         17            7           22          46   

   Percent of Total 2020 Country/Territory      Continent  2020 Population  \
0               0.105019     United States  North America        335942003   
1               0.082714             China           Asia       1424929781   
2               0.053903             Japan           Asia        125244761   
3               0.059480    United Kingdom         Europe         67059474   
4               0.042751         Australia        Oceania         25670051   

   Medals Per 10 Mil People  
0                  3.363676  
1                  0.624592  
2                  4.630932  
3                  9.543767  
4                 17.919715

I'm curious which countries have the most medals per capita!

In [18]:
data2020.sort_values('Medals Per 10 Mil People', ascending=False).head(10)

NOC  Gold 2020  Silver 2020  Bronze 2020  Total 2020  \
48  SMR          0            1            2           3   
9   NZL          7            6            7          20   
17  JAM          4            1            4           9   
55  BRN          0            1            0           1   
23  GEO          2            5            1           8   
11  HUN          6            7            7          20   
4   AUS         17            7           22          46   
40  EST          1            0            1           2   
16  NOR          4            2            2           8   
46  ARM          0            2            2           4   

    Percent of Total 2020 Country/Territory      Continent  2020 Population  \
48               0.002788        San Marino         Europe            34007   
9                0.018587       New Zealand        Oceania          5061133   
17               0.008364           Jamaica  North America          2820436   
55               0.000929            Brunei           Asia           441725   
23               0.007435           Georgia           Asia          3765912   
11               0.018587           Hungary         Europe          9750573   
4                0.042751         Australia        Oceania         25670051   
40               0.001859           Estonia         Europe          1329444   
16               0.007435            Norway         Europe          5379839   
46               0.003717           Armenia           Asia          2805608   

    Medals Per 10 Mil People  
48                882.171318  
9                  39.516843  
17                 31.909960  
55                 22.638519  
23                 21.243194  
11                 20.511615  
4                  17.919715  
40                 15.043883  
16                 14.870333  
46                 14.257159

So San Marino, New Zealand, Jamaica, Brunei, and Georgia got a disproportionately large number of medals in 2020! Good for them. I'm now curious how the different continents compare in terms of medal count.

In [19]:
data2020.groupby('Continent').agg({'Total 2020': 'sum'}).sort_values('Total 2020', ascending=False).reset_index()

Continent  Total 2020
0         Europe         287
1           Asia         238
2  North America         170
3        Oceania          66
4  South America          36
5         Africa          30

So Europe got the most medals overall, with Asia close behind. The last thing I'm curious to check is which continent is earning the most medals per capita. I'll do a similar process to find that out.

In [20]:
continent_data = data2020.groupby('Continent').agg({'Total 2020':'sum','2020 Population':'sum'}).reset_index()
continent_data['Medals Per 10 Mil People'] = (continent_data['Total 2020']/continent_data['2020 Population'])*10000000
continent_data.sort_values('Medals Per 10 Mil People', ascending=False)

Continent  Total 2020  2020 Population  Medals Per 10 Mil People
4        Oceania          66         30731184                 21.476556
2         Europe         287        438987461                  6.537772
3  North America         170        524949808                  3.238405
5  South America          36        355242046                  1.013394
1           Asia         238       3278459517                  0.725951
0         Africa          30        431378220                  0.695445

According to the table above, Oceania gets far more medals for its size, followed by Europe and North America.